front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/formatConf.ts

```diff
@@ -3,23 +3,37 @@ import type { Dispatch } from 'redux';
 
 import type { PowerRestrictionRange } from 'applications/operationalStudies/consts';
 import { formatAllowances } from 'modules/trainschedule/components/ManageTrainSchedule/Allowances/helpers';
-import { NO_POWER_RESTRICTION } from 'modules/trainschedule/components/ManageTrainSchedule/PowerRestrictionsSelector';
 import { setFailure } from 'reducers/main';
 import type { OsrdConfState } from 'reducers/osrdconf/consts';
 import { kmhToMs } from 'utils/physics';
 import { time2sec } from 'utils/timeManipulation';
 
+function mergeConsistentIdenticalObjects(data: PowerRestrictionRange[]) {
+  return data.reduce((mergedArray: PowerRestrictionRange[], currentObj, index, array) => {
+    if (index > 0 && currentObj.value === array[index - 1].value) {
+      const mergedObj = {
+        begin: array[index - 1].begin,
+        end: currentObj.end,
+        value: currentObj.value,
+      };
+      const previousObj = { ...mergedArray[mergedArray.length - 1] };
+      previousObj.end = mergedObj.end;
+      mergedArray[mergedArray.length - 1] = previousObj;
+      return mergedArray;
+    }
+    return mergedArray.concat(currentObj);
+  }, []);
+}
```

Comment:
Nice job, some remarks on this function:
- rename it so it's more explicit: mergeConsistentIdenticalObjects -> mergeIdenticalNeighbours ? and maybe add a comment to explain what this function does
- rename the variables: data -> powerRestrictionRanges, currentObj -> currentRange, mergedObj -> mergedRange, previousObj -> previousRange
- use early return: if index === 0 or prevValue !== currentValue, directly return the array with the new value at the end
- use a variable to store the previous value : prevRange = result[result.length - 1] and use it rather than the previous element in array
- use the function replaceElementAtIndex from utils/array rather than modify mergedArray (l21). In js, we avoid modifying an existing object, it's better to create a new one
- the new previous object has only its end changing, so you can create it like this: 
  ```
      const newPreviousRange = {
        ...previousRange,
        end: currentObj.end,
      };
  ```